elytron/src/test/java/org/wildfly/extension/elytron/TlsTestCase.java

```diff
@@ -60,15 +86,193 @@
+        Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
+
+        SelfSignedX509CertificateAndSigningKey muneraSelfSignedX509CertificateAndSigningKey = SelfSignedX509CertificateAndSigningKey.builder()
+                .setDn(MUNERASOFT_DN)
+                .setKeyAlgorithmName("RSA")
+                .setSignatureAlgorithmName("SHA256withRSA")
+                .addExtension(false, "BasicConstraints", "CA:true,pathlen:2147483647")
+                .build();
+        X509Certificate muneraCertificate = muneraSelfSignedX509CertificateAndSigningKey.getSelfSignedCertificate();
+
+        Calendar calendar = Calendar.getInstance();
+        Date currentDate = calendar.getTime();
+        calendar.add(Calendar.YEAR, 1);
+        Date nextYear = calendar.getTime();
+        calendar.add(Calendar.YEAR, -1);
+        calendar.add(Calendar.SECOND, -30);
+        Date revokeDate = calendar.getTime();
+
+        X509v2CRLBuilder crlBuilder = new X509v2CRLBuilder(
+                new X500Name(MUNERASOFT_DN.getName()),
+                currentDate
+        );
+        crlBuilder.addExtension(
+                Extension.authorityKeyIdentifier, false, new JcaX509ExtensionUtils().createAuthorityKeyIdentifier(muneraCertificate.getPublicKey())
+        );
+        crlBuilder.addExtension(
+                Extension.cRLNumber, false, new CRLNumber(BigInteger.valueOf(4110))
+        );
+        crlBuilder.addCRLEntry(
+                new BigInteger("1005"),
+                revokeDate,
+                CRLReason.unspecified
+        );
+        crlBuilder.addCRLEntry(
+                new BigInteger("1006"),
+                revokeDate,
+                CRLReason.unspecified
+        );
+        return crlBuilder.setNextUpdate(nextYear).build(
+                new JcaContentSignerBuilder("SHA256withRSA")
+                        .setProvider("BC")
+                        .build(muneraSelfSignedX509CertificateAndSigningKey.getSigningKey())
+        );
+    }
+
+    private static void createTemporaryCRLFile(X509CRLHolder crlHolder, File outputFile) throws Exception {
+        try (PemWriter output = new PemWriter(new OutputStreamWriter(new FileOutputStream(outputFile)))){
+            output.writeObject(new MiscPEMGenerator(crlHolder));
+        }
+    }
+
+    private static void createTemporaryKeyStoreFile(KeyStore keyStore, File outputFile) throws Exception {
+        try (FileOutputStream fos = new FileOutputStream(outputFile)){
+            keyStore.store(fos, GENERATED_KEYSTORE_PASSWORD);
+        }
+    }
+
+    private static void setUpKeyStores() throws Exception {
+        File workingDir = new File(WORKING_DIRECTORY_LOCATION);
+        if (workingDir.exists() == false) {
+            workingDir.mkdirs();
+        }
+
+        SelfSignedX509CertificateAndSigningKey issuerSelfSignedX509CertificateAndSigningKey = createIssuer();
+        File trustFile = new File(workingDir, "ca.truststore");
```

Comment:
Same thing here with the File declarations. Static variables could be used and referenced in this method and in ```deleteKeyStoreFiles()```.

Reply:
Fixed